benchmark_runner/common/oc/oc.py

```diff
@@ -1348,3 +1374,42 @@ def extract_vm_results(self, vm_name: str = '', start_stamp: str = '', end_stamp
                             # filter the data, placed after the first :
                             results_list.append(line.strip().split(':')[data_index:])
         return results_list
+
+    @typechecked
+    @logger_time_stamp
+    def generate_odf_must_gather(self, destination_path: str = '/tmp', odf_version: str = None):
+        """
+        Generates ODF must-gather logs based on the ODF version and stores it in the destination path.
+        :param destination_path: The directory where the must-gather logs will be stored. Default is '/tmp'.
+        :param odf_version: The version of ODF for which to generate the must-gather logs, Default is None.
+        :return: The result of the run command.
+        :raises: RuntimeError if the command fails.
+        """
+        folder_path = f"{destination_path}/odf-must-gather-rhel9-v{odf_version}"
+        try:
+            command = f"oc adm must-gather --image=registry.redhat.io/odf4/odf-must-gather-rhel9:v{odf_version} --dest-dir={folder_path}"
+            self.run(command)
+        except Exception as e:
+            if os.path.exists(folder_path):
+                shutil.rmtree(folder_path)
```

Comment:
Protect this so that if the rmtree fails we get the correct RuntimeError rather than whatever `shutil.rmtree` raises.